From 654c47ce460198e712eb8d1c34af0700e616c592 Mon Sep 17 00:00:00 2001
From: Jade Lovelace <lists@jade.fyi>
Date: Sat, 7 Jan 2023 09:58:26 -0500
Subject: [PATCH 2/4] plugins/iwd: Fix typo in create_known_network name

Signed-off-by: Jade Lovelace <lists@jade.fyi>
---
 plugins/iwd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/iwd.c b/plugins/iwd.c
index 5593fa74..9d437b3d 100644
--- a/plugins/iwd.c
+++ b/plugins/iwd.c
@@ -1714,7 +1714,7 @@ static void init_auto_connect(struct iwd_known_network *iwdkn)
 	}
 }
 
-static void create_know_network(GDBusProxy *proxy)
+static void create_known_network(GDBusProxy *proxy)
 {
 	const char *path = g_dbus_proxy_get_path(proxy);
 	struct iwd_known_network *iwdkn;
@@ -1825,7 +1825,7 @@ static void object_added(GDBusProxy *proxy, void *user_data)
 	else if (!strcmp(interface, IWD_NETWORK_INTERFACE))
 		create_network(proxy);
 	else if (!strcmp(interface, IWD_KNOWN_NETWORK_INTERFACE))
-		create_know_network(proxy);
+		create_known_network(proxy);
 	else if (!strcmp(interface, IWD_STATION_INTERFACE))
 		create_station(proxy);
 	else if (!strcmp(interface, IWD_AP_INTERFACE))
-- 
2.38.1

